pilot/pkg/model/destination_rule.go

```diff
@@ -50,24 +50,33 @@ func (ps *PushContext) mergeDestinationRule(p *consolidatedDestRules, destRuleCo
 	}
 
 	if mdrList, exists := destRules[resolvedHost]; exists {
-		// `addRuleToProcessedDestRules` determines if the incoming destination rule would become a new unique entry in the processedDestRules list.
-		addRuleToProcessedDestRules := true
+		// `appendSeparately` determines if the incoming destination rule would become a new unique entry in the processedDestRules list.
```

Comment:
Can we add a feature flag? Given all the breakages here, don't feel confident we can make changes like this safely.

Should be pretty easy to do